packages/perseus/src/widgets/radio/base-radio.jsx

```diff
@@ -468,8 +431,9 @@ class BaseRadio extends React.Component<$FlowFixMe, $FlowFixMe> {
                 <ul className={className} style={{listStyle: "none"}}>
                     {this.props.choices.map(function (choice, i) {
                         let Element = Choice;
+                        const ref = React.createRef();
+                        this.choiceRefs[i] = ref;
                         const elementProps = {
```

Comment:
**commentary:** I'm amazed that Flow is okay with this given that the `Choice` and `ChoiceAndNone` have slightly different prop types.

Reply:
Further down it changes the props based on which component it's going to use:

```
if (choice.isNoneOfTheAbove) {
    Element = ChoiceNoneAbove;
    _.extend(elementProps, {
        showContent: choice.revealNoneOfTheAbove,
    });
}
```

Not sure if that clears anything up.

Reply:
Right, but after that `elementProps.showContent` is `undefined | typeof choice.revealNoneOfTheAbove`.